### PR TITLE
feat: Search for Hashtags

### LINKED ISSUE #2143: 🚀 [Feature] Search by hashtags
> [alina-beck](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-06T14:31:11Z" title="Wednesday, November 6th 2019, 3:31:11 pm +01:00">Nov 6, 2019</time>_
_Closed <time datetime="2020-03-20T11:25:48Z" title="Friday, March 20th 2020, 12:25:48 pm +01:00">Mar 20, 2020</time>_
---

## :rocket: Feature
Users should be able to search by hashtags by entering a keyword in the search field.

### User Problem
At the moment users can only find posts by hashtag by navigating to a post and clicking the hashtag in its content. This takes a long time and can be very frustrating, especially since they first need to _find_ a post that contains the hashtag they are interested in.

### Implementation
The list of search results should include a section with `hashtags` where the relevant hashtags are listed. Selecting a hashtag initiates the hashtag search the way it is working now (i.e. displaying the hashtag on top of a page with posts containing that hashtag).

### Design & Layout
The hashtag results should look similar to the post and user results in this example:

### Additional context
This is **Step 3** in our search implementation – see #2146.
**Step 1** is adding users to the list of search results: #1463.
**Step 2** is limiting the number of search results to 15: #2142.

related to #1068

### PR DESCRIPTION
**Pull request by Tirokk**
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-19T12:10:48Z" title="Thursday, March 19th 2020, 1:10:48 pm +01:00">Mar 19, 2020</time>_
_Merged <time datetime="2020-03-20T11:25:48Z" title="Friday, March 20th 2020, 12:25:48 pm +01:00">Mar 20, 2020</time>_
---

## 🍰 Pullrequest
Adding the search results for hashtags to the drop down menu

- fixes #2143

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-03-19T13:10:42Z" title="Thursday, March 19th 2020, 2:10:42 pm +01:00">Mar 19, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6243/specs"></a> <b title="Passed">66</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6243/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 100a5cdec5 <span style="cursor:help;" title="Your CI provider generated this commit by merging this branch into the target branch.">ℹ️</span> |
| **Started** | Mar 19, 2020 7:35 PM |
| **Ended** | Mar 19, 2020 7:55 PM |
| **Duration** | 20:43 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Firefox 68  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/6243/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-03-19T13:12:26Z" title="Thursday, March 19th 2020, 2:12:26 pm +01:00">Mar 19, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6247/specs"></a> <b title="Passed">66</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/6247/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | fa7f1bb111 |
| **Started** | Mar 19, 2020 8:23 PM |
| **Ended** | Mar 19, 2020 8:47 PM |
| **Duration** | 23:39 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Firefox 68  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/6247/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: backend/src/db/migrations/20200207080200-fulltext_index_for_tags.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export async function down(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+      CALL db.index.fulltext.drop("tag_fulltext_search")
+    `)
```
Comment (Tirokk) [inline backend/src/db/migrations/20200207080200-fulltext_index_for_tags.js:38]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-20T11:23:32Z" title="Friday, March 20th 2020, 12:23:32 pm +01:00">Mar 20, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,51 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description =
+  'This migration adds a fulltext index for the tags in order to search for Hasthags.'
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    await transaction.run(`
+      CALL db.index.fulltext.createNodeIndex("tag_fulltext_search",["Tag"],["id"])
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
+    throw new Error(error)
+  } finally {
+    session.close()
+  }
+}
+
+export async function down(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+      CALL db.index.fulltext.drop("tag_fulltext_search")
+    `)
```


I really like that you are adding the `down` function as well. I think I haven't added it on every migration, and some are easier to add then others, but just thought I'd point out that this is a great habit!

**Review by Mogge — CHANGES_REQUESTED** [review]
hey, it looks good! If it were me I would probably look for use this as an opportunity to rename `Hashtag.vue` to something else, and maybe even node type `Tag` to `Hashtag`, but it's debatable whether that should happen in this `PR`... 
I would refactor the hashtags watch :point_down: though ...

**Review by Mogge — APPROVED** [review]
great job @Mogge !!
thanks for this, I'm sure the community will love it!!

**Pull request closed without merging**